mitodl/ocw-studio
# Website CRUD permissions

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #33 

#### What's this PR do?
- Sets up groups and django-guardian object-specific permissions to determine who should be able to CRUD websites and website content, publish websites, and assign collaborators (admins, editors).

#### How should this be manually tested?
- Migrate, then run `manage.py backpopulate_groups` if you've already imported/created some websites.
Tests should pass.  There are no views yet that use these permissions but you could experiment with them in a shell following the unit tests as examples.  If you go to the django admin edit page for.a Website, there will be a "Object Permissions" button near the top right.  Clicking it should show you this:

<img width="1061" alt="Screen Shot 2021-02-05 at 3 25 44 PM" src="https://user-images.githubusercontent.com/187676/107085544-a424e900-67c6-11eb-9497-ab0be521ce7f.png">


#### Any background context you want to provide?

- For creating and deleting WebsiteContent, I made the assumption that we will use [nested routes/views](https://chibisov.github.io/drf-extensions/docs/#nested-routes) provided by drf-extensions - this is how a user's course list items are handled in open-discussions.


#### What GIF best describes this PR or how it makes you feel?
![gandalf](https://user-images.githubusercontent.com/187676/106903196-b2d8a680-66c7-11eb-8003-9563a9a8c8c5.gif)
